### DISCUSSION TITLE
Block names

### DISCUSSION THREAD

**Discussion by styyokuda** [idaholab/moose#25952] [Q&A Meshing]
All, 
I am using "CartesianMeshGenerator" to generate blocks and mesh.
"CartesianMeshGenerator" seems to provide only block numbers.
Is there any way to put names on generated blocks.
Please let me know.
S. Thomas

**Comment by GiudGiud** [comment]
Hello

You can use the RenameBlockGenerator to do that
https://mooseframework.inl.gov/source/meshgenerators/RenameBlockGenerator.html

Guillaume

**Reply by styyokuda** [reply]
Thank you, Guillaume.

I thought that "RenameBlockGenerator" changes the block numbers to block names.
I have tried it and got some strange things. Probably, I did something wrong.

Does "RenameBlockGenerator" keep both the block numbers and the block names?
Please help.
S. Thomas

**Reply by GiudGiud** [reply]
It can change the number too
You should use names uniformly across the input though, it s the recommended moose way.
